runtime.go

```diff
@@ -1,53 +1,6 @@
 package containerd
 
-import (
-	"fmt"
-	"sync"
-
-	"golang.org/x/net/context"
-)
-
-// NewRuntimeFunc is the runtime's constructor
-type NewRuntimeFunc func(root string) (Runtime, error)
-
-var runtimeRegistration = struct {
-	mu       sync.Mutex
-	runtimes map[string]NewRuntimeFunc
-}{
-	runtimes: make(map[string]NewRuntimeFunc),
-}
-
-// RegisterRuntime is not external packages registers Runtimes for use with containerd
-func RegisterRuntime(name string, f NewRuntimeFunc) {
-	runtimeRegistration.mu.Lock()
-	defer runtimeRegistration.mu.Unlock()
-	if _, ok := runtimeRegistration.runtimes[name]; ok {
-		panic(fmt.Errorf("runtime already registered as %q", name))
-	}
-	runtimeRegistration.runtimes[name] = f
-}
-
-// Runtimes returns a slice of all registered runtime names for containerd
-func Runtimes() (o []string) {
-	runtimeRegistration.mu.Lock()
-	defer runtimeRegistration.mu.Unlock()
-
-	for k := range runtimeRegistration.runtimes {
-		o = append(o, k)
-	}
-	return o
-}
-
-// NewRuntime calls the runtime's constructor with the provided root
-func NewRuntime(name, root string) (Runtime, error) {
-	runtimeRegistration.mu.Lock()
-	defer runtimeRegistration.mu.Unlock()
-	f, ok := runtimeRegistration.runtimes[name]
-	if !ok {
-		return nil, ErrRuntimeNotExist
-	}
-	return f(root)
-}
+import "golang.org/x/net/context"
```

Comment:
we can do an update in another pr changing all of the places that contexts are used